### PR TITLE
Use DOMMatrix2DInit as parameters for methods. #326

### PR DESCRIPTION
**Pull request by dirkschulze**

### CODE REVIEW

File: master/coords.html [hunk 1]
```diff
@@ -3226,7 +3226,7 @@ <h3 id="InterfaceSVGTransform">Interface SVGTransform</h3>
   [SameObject] readonly attribute <a>DOMMatrix</a> <a href="coords.html#__svg__SVGTransform__matrix">matrix</a>;
   readonly attribute float <a href="coords.html#__svg__SVGTransform__angle">angle</a>;
 
-  void <a href="coords.html#__svg__SVGTransform__setMatrix">setMatrix</a>(<a>DOMMatrixReadOnly</a> matrix);
+  void <a href="coords.html#__svg__SVGTransform__setMatrix">setMatrix</a>(optional <a>DOMMatrix2DInit</a> matrix);
```
Comment (AmeliaBR) [inline master/coords.html]: Is the default value defined somewhere? Does a missing dictionary default to all zeros, or to an identity matrix? 2D or 3D?

**Comment by dirkschulze** [comment]
@AmeliaBR ping?

**Comment by AmeliaBR** [comment]
Sorry! Reviewing the linked geometry interfaces text, I think it safely covers the default behavior.

However, the way you have written the algorithm, I don't think you include the full error handling behavior from the validate and fix-up method, because you only run the steps as part of a hypothetical test for `is2D()`.

I would rewrite the steps as follows:

<ol class='algorithm'>
  <li>If the <a>SVGTransform</a> object is <a href='#ReadOnlyTransform'>read only</a>, then
  <a>throw</a> a <a>NoModificationAllowedError</a>.</li>
  <li> Let <var>newMatrix</var> be the result of <code><a>DOMMatrixReadOnly</a>.fromMatrix(<var>matrix</var>)</code>,
  including the <a href="https://drafts.fxtf.org/geometry/#dommatrixinit-dictionary">validate and fix-up</a> steps for missing values.
  If that method throws an error, then re-throw that error and abort these steps.</li>
  <li>If <var>newMatrix</var>.is2D() would return true,
  then set the <a>SVGTransform</a> object's
  <a href='#TransformValue'>value</a> to a <span class='prop-value'>matrix(…)</span>
  value that represents the same matrix as <var>newMatrix</var>.</li>
  <li>Otherwise, set the <a>SVGTransform</a> object's
  <a href='#TransformValue'>value</a> to a <span class='prop-value'>matrix3d(…)</span>
  value that represents the same matrix as <var>newMatrix</var>.</li>
  <li>In either case, set <var>newMatrix</var>
  as the <a>SVGTransform</a> object's <a href="#TransformMatrixObject">matrix object</a>.</li>
  <li>If the <a>SVGTransform</a> object
  <a href="#TransformMode">reflects a presentation attribute value of an element</a>,
  then <a>reserialize</a> the reflected attribute.</li>
</ol>

```html
<ol class='algorithm'>
  <li>If the <a>SVGTransform</a> object is <a href='#ReadOnlyTransform'>read only</a>, then
  <a>throw</a> a <a>NoModificationAllowedError</a>.</li>
  <li> Let <var>newMatrix</var> be the result of <code><a>DOMMatrixReadOnly</a>.fromMatrix(<var>matrix</var>)</code>,
  including the <a href="https://drafts.fxtf.org/geometry/#dommatrixinit-dictionary">validate and fix-up</a> steps for missing values.
  If that method throws an error, then re-throw that error and abort these steps.</li>
  <li>If <var>newMatrix</var>.is2D() would return true,
  then set the <a>SVGTransform</a> object's
  <a href='#TransformValue'>value</a> to a <span class='prop-value'>matrix(…)</span>
  value that represents the same matrix as <var>newMatrix</var>.</li>
  <li>Otherwise, set the <a>SVGTransform</a> object's
  <a href='#TransformValue'>value</a> to a <span class='prop-value'>matrix3d(…)</span>
  value that represents the same matrix as <var>newMatrix</var>.</li>
  <li>In either case, set <var>newMatrix</var>
  as the <a>SVGTransform</a> object's <a href="#TransformMatrixObject">matrix object</a>.</li>
  <li>If the <a>SVGTransform</a> object
  <a href="#TransformMode">reflects a presentation attribute value of an element</a>,
  then <a>reserialize</a> the reflected attribute.</li>
</ol>
```

**Comment by AmeliaBR** [comment]
@dirkschulze Ok, those points make sense, too. My main point was that we actually have to run the full validate & fix-up algorithm to adjust the input value before using it, and throw any errors it issues.

Can you take those ideas and come up with a better edit?

**Comment by AmeliaBR** [comment]
Ah, realized one thing is missing. This is a normative change to the API, so there should be a changes appendix entry.

**Pull request merged**